macros/utils/table_operations/insert.sql

```diff
@@ -9,31 +9,52 @@
         {{ elementary.edr_log('Could not extract columns for table - ' ~ table_name ~ ' (might be a permissions issue)') }}
         {{ return(none) }}
     {% endif %}
-    {% set dicts_chunks = elementary.split_list_to_chunks(dicts, chunk_size) %}
-    {% for dicts_chunk in dicts_chunks %}
-        {% set insert_dicts_query = elementary.get_insert_dicts_query(table_relation, columns, dicts_chunk) %}
+    {% for dicts_chunk in dicts | batch(chunk_size) %}
+        {% set insert_dicts_query = elementary.get_insert_rows_query(table_relation, columns, dicts_chunk) %}
         {% do run_query(insert_dicts_query) %}
     {% endfor %}
     {%- if should_commit -%}
         {% do adapter.commit() %}
     {%- endif -%}
 {% endmacro %}
 
-{% macro get_insert_dicts_query(table_relation, columns, dicts) -%}
-    {% set insert_dicts_query %}
+
+{% macro insert_table(table_relation, table, chunk_size=10000, should_commit=False) %}
```

Comment:
Not sure why we need this, could we maybe use insert dicts and pass the agate rows to insert_dicts?

Reply:
We duplicate your memory and waste compute by iterating over an agate's rows and copying them into dicts?
This is only a thin wrapper, it does the same thing.

Reply:
Are you sure that passing the rows to insert_dicts means copying the dicts?

Reply:
In order to use `insert_dicts()` you need to use `agate_to_dicts` which clones the same data and by that duplicates it in memory, so yes.

Reply:
I now see that `insert_dicts` works with `agate.Table`.
Let's just rename `insert_dicts` to `insert_rows` to indicate that it works with all row-iterables.
What do you think?